pandas/core/internals/blocks.py

```diff
@@ -1421,6 +1421,32 @@ def quantile(
         result = ensure_block_shape(result, ndim=2)
         return new_block_2d(result, placement=self._mgr_locs)
 
+    def round(self, decimals: int, using_cow: bool = False) -> Block:
+        """
+        Rounds the values.
+        If the block is not of an integer or float dtype, nothing happens.
+        This is consistent with DataFrame.round behavivor.
+        (Note: Series.round would raise)
+
+        Parameters
+        ----------
+        decimals: int,
+            Number of decimal places to round to.
+            Caller is responsible for validating this
+        using_cow: bool,
+            Whether Copy on Write is enabled right now
+        """
+        if not self.is_numeric or self.is_bool:
+            return self.copy(deep=not using_cow)
+        # TODO: round only defined on BaseMaskedArray
+        # Series also does this, so would need to fix both places
+        # error: Item "ExtensionArray" of "Union[ndarray[Any, Any], ExtensionArray]"
+        # has no attribute "round"
+        return new_block_2d(
+            self.values.round(decimals),  # type: ignore[union-attr]
```

Comment:
I think you want to use ``self.make_block_same_class`` here? If class could change then ``self.make_block``. Don't have to set placement there

Reply:
This is dangerous, I don't think that round guarantees a copy, e.g. 

```
arr = np.array([1, 2])
arr2 = arr.round(2)
arr2[0] = 100
```

You can check this with

```
arr is arr2
```

(also please add a test for this case)

Reply:
Thanks for the tip.
I found the source here https://github.com/numpy/numpy/blob/486878b37fc7439a3b2b87747f50db9b62fea8eb/numpy/core/src/multiarray/calculation.c#L625-L636
and it looks like this is only triggered for integer case with decimals >= 0.

BTW, it also looks like round is an operation that can be done ``inplace`` by using the ``out`` parameter in numpy.
Maybe I should open a PR for that as a follow-up?

Reply:
Would have to check if it's actually faster, but if yes then should utilise it in CoW cases

Reply:
Follow up sounds good to me